pyproject.toml

```diff
@@ -31,10 +31,11 @@ dependencies = [
 [project.optional-dependencies]
 dev = [
     "mypy",
-    "ruff",
+    "ruff==0.3.4",
     "pytest",
     "types-pyyaml",
     "pip-tools",
+    "pre-commit"
 ]
 
```

Comment:
Need to add `pynxtools` in 
```
consistency_with_pynxtools = [
    "pynxtools>=0.1.1",
]
```
under the `optional-dependencies`. Otherwise, it is almost impossible to run the test in the local dev env.